xmtp_api_d14n/src/compat/d14n.rs

```diff
@@ -41,80 +48,318 @@ where
     E: std::error::Error + RetryableError + Send + Sync + 'static,
     P: Send + Sync + Client,
     C: Send + Sync + Client,
+    ApiError<E>: From<ApiError<<P as Client>::Error>>
+        + From<ApiError<<C as Client>::Error>>
+        + Send
+        + Sync
+        + 'static,
+    ApiError<E>: From<GrpcError>,
```

Comment:
we can't depend on GrpcError inside this crate, since we want to use it for both grpc and http apis. You need to create a new error variant to cover the cases for this.

if the conversions are covered in replication client, they should be moved to xmtp_api_d14n, and the error type should bev replaced with something generic over http or grpc. 


This shouldn't be a big deal, since the conversions here dont depend on http or grpc, but instead are plain type conversion, which usually error on missing fields or something

Reply:
Certainly, just wanted to get rid of build issues now. Will fix it.